### PR TITLE
xref: Add bindings for xref--transient-buffer-mode

### PR DESCRIPTION
**Pull request by condy0919**

### CODE REVIEW

File: modes/xref/evil-collection-xref.el [hunk 1]
```diff
@@ -50,13 +52,24 @@
...
     "go" 'xref-show-location-at-point)
 
   (when (>= emacs-major-version 27)
     (evil-collection-define-key 'normal 'xref--xref-buffer-mode-map
-      "gr" 'xref-revert-buffer)))
+      "gr" 'xref-revert-buffer))
+
+  (when (>= emacs-major-version 28)
+    (evil-collection-define-key 'normal 'xref--xref-buffer-mode-map
+      "[[" 'xref-prev-group
```
Comment (jojojames) [inline modes/xref/evil-collection-xref.el:66]: I wonder if this conflicts with unimpaired style bindings.

File: modes/xref/evil-collection-xref.el [hunk 2]
```diff
@@ -50,13 +52,24 @@
 
     ;; open
     (kbd "RET") 'xref-goto-xref
+    (kbd "TAB") 'xref-quit-and-goto-xref
```
Comment (jojojames) [inline modes/xref/evil-collection-xref.el:55]: This one might be unexpected by a user. Maybe another key?

Comment (condy0919) [reply modes/xref/evil-collection-xref.el:55]: That's the default Emacs style binding:

``` elisp
(defvar xref--xref-buffer-mode-map
  (let ((map (make-sparse-keymap)))
    (define-key map (kbd "n") #'xref-next-line)
    (define-key map (kbd "p") #'xref-prev-line)
    (define-key map (kbd "N") #'xref-next-group)
    (define-key map (kbd "P") #'xref-prev-group)
    (define-key map (kbd "r") #'xref-query-replace-in-results)
    (define-key map (kbd "RET") #'xref-goto-xref)
    (define-key map (kbd "TAB")  #'xref-quit-and-goto-xref)
    (define-key map (kbd "C-o") #'xref-show-location-at-point)
    ;; suggested by Johan Claesson "to further reduce finger movement":
    (define-key map (kbd ".") #'xref-next-line)
    (define-key map (kbd ",") #'xref-prev-line)
    (define-key map (kbd "g") #'xref-revert-buffer)
    map))
```

Agree that it's confused, what about changing to `S-RET`?
`xref-quit-and-goto-xref` has already two bindings (`o` and `go`) to invoke.

File: modes/xref/evil-collection-xref.el (hunk 1 above)
Comment (condy0919) [reply modes/xref/evil-collection-xref.el:66]: As far as I known, unimpaired won't/shouldn't use `[[` and `]]`. Those are used in vim motion state.

File: modes/xref/evil-collection-xref.el (hunk 2 above)
Comment (jojojames) [reply modes/xref/evil-collection-xref.el:55]: Maybe it's OK, I'm not sure if we have precedence yet for "action -> close current window".

**Comment by condy0919** [comment]
@jojojames ping

**Pull request merged**